Support module for JUnit tests

The test always fails when using "module-info.java", but `mvn test` works fine. it took me hours to find the cause, seems no one has this problem, can anyone help me?

<img src="https://user-images.githubusercontent.com/32408633/64945651-9b670680-d8a3-11e9-8939-89cb6d54d278.png" width="400">

<img src="https://user-images.githubusercontent.com/32408633/64945589-72df0c80-d8a3-11e9-9071-b9eb7a02063f.png" width="300">

```
9月 16, 2019 4:49:14 下午 org.junit.platform.launcher.core.DefaultLauncher handleThrowable
WARNING: TestEngine with ID 'junit-vintage' failed to discover tests
org.junit.platform.commons.PreconditionViolationException: Could not load class with name: com.tester.EmptyServiceTest
	at org.junit.platform.engine.discovery.MethodSelector.lambda$lazyLoadJavaClass$0(MethodSelector.java:157)
	at org.junit.platform.commons.function.Try$Failure.getOrThrow(Try.java:335)
	at org.junit.platform.engine.discovery.MethodSelector.lazyLoadJavaClass(MethodSelector.java:156)
	at org.junit.platform.engine.discovery.MethodSelector.getJavaClass(MethodSelector.java:135)
	at org.junit.vintage.engine.discovery.MethodSelectorResolver.resolve(MethodSelectorResolver.java:34)
	at org.junit.vintage.engine.discovery.MethodSelectorResolver.lambda$resolve$0(MethodSelectorResolver.java:30)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.vintage.engine.discovery.MethodSelectorResolver.resolve(MethodSelectorResolver.java:30)
	at org.junit.vintage.engine.discovery.VintageDiscoverer.collectTestClasses(VintageDiscoverer.java:64)
	at org.junit.vintage.engine.discovery.VintageDiscoverer.discover(VintageDiscoverer.java:51)
	at org.junit.vintage.engine.VintageTestEngine.discover(VintageTestEngine.java:61)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverEngineRoot(DefaultLauncher.java:168)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverRoot(DefaultLauncher.java:155)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
	at com.microsoft.java.test.runner.junit5.CustomizedConsoleTestExecutor.lambda$executeTests$0(CustomizedConsoleTestExecutor.java:42)
	at com.microsoft.java.test.runner.junit5.CustomContextClassLoaderExecutor.invoke(CustomContextClassLoaderExecutor.java:29)
	at com.microsoft.java.test.runner.junit5.CustomizedConsoleTestExecutor.executeTests(CustomizedConsoleTestExecutor.java:38)
	at com.microsoft.java.test.runner.junit5.CustomizedConsoleLauncher.execute(CustomizedConsoleLauncher.java:29)
	at com.microsoft.java.test.runner.Launcher.main(Launcher.java:56)
Caused by: java.lang.ClassNotFoundException: com.tester.EmptyServiceTest
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
	at org.junit.platform.commons.util.ReflectionUtils.lambda$tryToLoadClass$9(ReflectionUtils.java:779)
	at org.junit.platform.commons.function.Try.lambda$call$0(Try.java:57)
	at org.junit.platform.commons.function.Try.of(Try.java:93)
	at org.junit.platform.commons.function.Try.call(Try.java:57)
	at org.junit.platform.commons.util.ReflectionUtils.tryToLoadClass(ReflectionUtils.java:742)
	at org.junit.platform.commons.util.ReflectionUtils.tryToLoadClass(ReflectionUtils.java:698)
	... 17 more
```
[github source](https://github.com/jimliuxyz/junittest)

--- a/java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/launchers/JUnitLaunchConfigurationDelegate.java
+++ b/java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/launchers/JUnitLaunchConfigurationDelegate.java
@@ -14,6 +14,7 @@ package com.microsoft.java.test.plugin.launchers;
 import org.eclipse.core.runtime.CoreException;
 import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.debug.core.DebugPlugin;
 import org.eclipse.debug.core.ILaunch;
 import org.eclipse.debug.core.ILaunchConfiguration;
 import org.eclipse.debug.core.Launch;
@@ -22,6 +23,9 @@ import org.eclipse.jdt.launching.VMRunnerConfiguration;
 
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
 public class JUnitLaunchConfigurationDelegate extends org.eclipse.jdt.junit.launcher.JUnitLaunchConfigurationDelegate {
     public JUnitLaunchArguments getJUnitLaunchArguments(ILaunchConfiguration configuration, String mode,
@@ -42,7 +46,7 @@ public class JUnitLaunchConfigurationDelegate extends org.eclipse.jdt.junit.laun
             launchArguments.projectName = javaProject.getProject().getName();
             launchArguments.classpath = config.getClassPath();
             launchArguments.modulepath = config.getModulepath();
-            launchArguments.vmArguments = config.getVMArguments();
+            launchArguments.vmArguments = getVmArguments(config);
             launchArguments.programArguments = config.getProgramArguments();
 
             return launchArguments;
@@ -50,6 +54,23 @@ public class JUnitLaunchConfigurationDelegate extends org.eclipse.jdt.junit.laun
                 InvocationTargetException e) {
             return null;
         }
+    }
+
+    private String[] getVmArguments(VMRunnerConfiguration config) {
+        final List<String> vmArgs = new ArrayList<>();
+        vmArgs.addAll(Arrays.asList(config.getVMArguments()));
+        
+        if (config.isPreviewEnabled()) {
+            vmArgs.add("--enable-preview");
+        }
+
+        final String dependencies = config.getOverrideDependencies();
+        if (dependencies != null && dependencies.length() > 0) {
+            final String[] parseArguments = DebugPlugin.parseArguments(dependencies);
+            vmArgs.addAll(Arrays.asList(parseArguments));
+        }
+
+        return vmArgs.toArray(new String[vmArgs.size()]);
     }
 
     public static class JUnitLaunchArguments {

